File: invert-binary-tree.cpp
using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
		if (root == nullptr) return nullptr;

		auto new_left = invertTree(root->right);
		auto new_right = invertTree(root->left);
		root->left = new_left;
		root->right = new_right;
		return root;
    }
};

File: invert-binary-tree.py
from __future__ import annotations
from typing import Optional

class TreeNode:
    def __init__(self, val: int = 0, left: Optional[TreeNode] = None, right: Optional[TreeNode] = None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root is None:
            return root

        new_right = self.invertTree(root.left)
        new_left = self.invertTree(root.right)
        root.left = new_left
        root.right = new_right
        return root
